ingestion-lambda/src/categoryCodes.ts

```diff
@@ -40,19 +43,49 @@ export function processFingerpostAPCategoryCodes(original: string[]): string[] {
 	return deduped;
 }
 
-export function processFingerpostAAPCategoryCodes(categoryCodes: string[]): string[] {
-	const allCategoryCodes = categoryCodes
-		.flatMap((categoryCode) => categoryCode.split('|'))
+export function processFingerpostAAPCategoryCodes(
+	categoryCodes: string[],
+): string[] {
+	const allCategoryCodes = categoryCodes.flatMap((categoryCode) =>
+		categoryCode.split('|'),
+	);
 
-	const mediaTopics = allCategoryCodes
-		.filter((_) => _.split(':').length > 1)
+	const mediaTopics = allCategoryCodes.filter((_) => _.split(':').length > 1);
 
 	const legacySubjectCodes = allCategoryCodes
 		.filter((_) => _.split('+').length > 1)
 		.map((categoryCode) => {
-			const [ code, _label ] = categoryCode.split('+');
-			return `subj:${code}`
+			const [code, _label] = categoryCode.split('+');
+			return `subj:${code}`;
 		});
 
-	return [...mediaTopics, ...legacySubjectCodes]
+	return [...mediaTopics, ...legacySubjectCodes];
+}
+
+// example input: "iptccat:HUM+SCI"
+export function processFingerpostAFPCategoryCodes(
+	original: string[],
+): string[] {
+	function flattenCategoryCodes(categoryCodes: string): string[] {
+		const [prefix, ...codes] = categoryCodes.split(':');
+		return codes
+			.flatMap((_) => _.split('+'))
+			.filter((_) => _.trim().length > 0)
```

Comment:
The part is duplicated; it might be good to move it to a function:
```
codes
    .flatMap((_) => _.split('+'))
    .filter((_) => _.trim().length > 0)
```

Reply:
I did consider that, but then I was thinking that I'd follow the 'rule of three' and abstract it if we see a third application of the same pattern?